src/Suppressor.jl

```diff
@@ -182,6 +182,59 @@ macro capture_err(block)
     end
 end
 
+"""
+    @capture expr
+
+Capture the `output` and `stderr` streams for the given expression,
+return a tuple of stdout and stderr.
+"""
+macro capture(block)
+    quote
+        if ccall(:jl_generating_output, Cint, ()) == 0
+            original_stdout = stdout
+            out_rd, out_wr = redirect_stdout()
+            out_reader = @async read(out_rd, String)
+
+            original_stderr = stderr
+            err_rd, err_wr = redirect_stderr()
+            err_reader = @async read(err_rd, String)
+
+
+            
+            # approach adapted from https://github.com/JuliaLang/IJulia.jl/pull/667/files
+            logstate = Base.CoreLogging._global_logstate
+            logger = logstate.logger
+            if logger.stream == original_stderr
+                new_logstate = Base.CoreLogging.LogState(typeof(logger)(err_wr, logger.min_level))
+                Core.eval(Base.CoreLogging, Expr(:(=), :(_global_logstate), new_logstate))
```

Comment:
This hack seems quite ugly and fragile. It's quite likely to break in future versions as it relies on a lot of implementation details.

It also basically assumes that `logger` is a `Logging.ConsoleLogger` because it assumes that `logger.stream` exists.

I'm not sure what the intent of checking `logger.stream == original_stderr`, but other than that it would be better to create a new `ConsoleLogger`, and use the public API `Logging.with_logger` to install it. Then you can avoid relying on a big pile of implementation detail.

Reply:
Yeah this is pre-Julia-1.0 code! 

`with_logger` makes total sense, that's probably better than using `global_logger(logger)`

Reply:
Yeah, kind of more understandable:  https://github.com/fonsp/PlutoUI.jl/pull/31#issuecomment-691339057
However not sure if I can ignore @async ...